### PR TITLE
Winduprule 142 show spring remote services ( from XML ) on services report

### PR DESCRIPTION
**Pull request by jonathanvila**

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Can one of the admins verify this patch?

**Comment by windupgithubbot1** [comment]
Can one of the admins verify this patch?

**Comment by mrizzi** [comment]
retest

**Comment by mrizzi** [comment]
retest

**Comment by mrizzi** [comment]
retest

**Comment by mrizzi** [comment]
retest

**Comment by mrizzi** [comment]
retest

**Comment by mrizzi** [comment]
retest

**Comment by mrizzi** [comment]
retest

**Comment by mrizzi** [comment]
@windupgithubbot1 Test this please

**Comment by m-brophy** [comment]
retest

File: rules-java-ee/tests/src/test/resources/discover-spring-remote-services/RestController.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class EmployeeController {
```
Comment (m-brophy) [inline rules-java-ee/tests/src/test/resources/discover-spring-remote-services/RestController.java]: Shouldn't the file name of this class be EmployeeController? I believe test resource classes don't need to compile so it probably doesn't matter.

**Comment by m-brophy** [comment]
Hi @jonathanvila I've looked through all the code, made some comments and run the tests. The comments are questions rather than recommendations, if you're happy they don't require any changes then let me know and I'll approve this.

File: rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/CreateRemoteReportRuleProvider.java [hunk 2]
```diff
@@ -90,44 +95,79 @@ private void createReport(GraphContext context, ProjectModel projectModel)
...
+                    httpinvokerList.add(remoteServiceModel);
+
+                } else if (packageName.toLowerCase().contains(".jaxws")) {
+                    WebServiceModel webServiceModel = new JaxWSWebServiceModelService(context).getOrCreate(projectModel, ((SpringRemoteServiceModel) remoteServiceModel).getInterface(), ((SpringRemoteServiceModel) remoteServiceModel).getImplementationClass());
+                    jaxWsList.add(webServiceModel);
+                }
             }
         }
 
         if (jaxRsList.isEmpty() && jaxWsList.isEmpty() && ejbRemoteList.isEmpty() && rmiList.isEmpty())
```
[the remark this answers is not in the document]
Comment (jonathanvila) [reply rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/CreateRemoteReportRuleProvider.java]: +1

File: rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEjbAnnotationsRuleProvider.java [hunk 3]
```diff
@@ -114,7 +116,8 @@ private void extractMessageDrivenMetadata(GraphRewrite event, JavaTypeReferenceM
         javaTypeReference.getFile().setGenerateSourceReport(true);
         JavaAnnotationTypeReferenceModel annotationTypeReference = (JavaAnnotationTypeReferenceModel) javaTypeReference;
 
-        JavaClassModel ejbClass = getJavaClass(javaTypeReference);
+        JavaClassService javaClassService = new JavaClassService(event.getGraphContext());
+        JavaClassModel ejbClass = javaClassService.getJavaClass(javaTypeReference);
 
         String ejbName = getAnnotationLiteralValue(annotationTypeReference, "name");
```
[the remark this answers is not in the document]
Comment (jonathanvila) [reply rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverEjbAnnotationsRuleProvider.java]: +1

File: rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverJPAAnnotationsRuleProvider.java [hunk 4]
```diff
@@ -169,7 +170,8 @@ private void extractEntityBeanMetadata(GraphRewrite event, JavaTypeReferenceMode
         JavaAnnotationTypeReferenceModel entityAnnotationTypeReference = (JavaAnnotationTypeReferenceModel) entityTypeReference;
         JavaAnnotationTypeReferenceModel tableAnnotationTypeReference = findTableAnnotation(event, entityTypeReference);
 
-        JavaClassModel ejbClass = getJavaClass(entityTypeReference);
+        JavaClassService javaClassService = new JavaClassService(event.getGraphContext());
+        JavaClassModel ejbClass = javaClassService.getJavaClass(entityTypeReference);
 
         String ejbName = getAnnotationLiteralValue(entityAnnotationTypeReference, "name");
```
[the remark this answers is not in the document]
Comment (jonathanvila) [reply rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverJPAAnnotationsRuleProvider.java]: +1

File: rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverJaxWSAnnotationsRuleProvider.java [hunk 5]
```diff
@@ -64,7 +64,7 @@ private void extractMetadata(GraphRewrite event, JavaTypeReferenceModel typeRefe
     {
         JavaClassService javaClassService = new JavaClassService(event.getGraphContext());
 
-        JavaClassModel implementationClass = getJavaClass(typeReference);
+        JavaClassModel implementationClass = javaClassService.getJavaClass(typeReference);
 
         // first, find out if it implements an interface.
         // TODO: handle the interface only case, where clients exist but no implementation
```
[the remark this answers is not in the document]
Comment (jonathanvila) [reply rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverJaxWSAnnotationsRuleProvider.java]: +1

File: rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/datasource/DiscoverDataSourceAnnotationRuleProvider.java [hunk 6]
```diff
@@ -78,7 +79,8 @@ private void extractDataSourceMetadata(GraphRewrite event, JavaTypeReferenceMode
         javaTypeReference.getFile().setGenerateSourceReport(true);
         JavaAnnotationTypeReferenceModel annotationTypeReference = (JavaAnnotationTypeReferenceModel) javaTypeReference;
 
-        JavaClassModel datasourceClass = getJavaClass(javaTypeReference);
+        JavaClassService javaClassService = new JavaClassService(event.getGraphContext());
+        JavaClassModel datasourceClass = javaClassService.getJavaClass(javaTypeReference);
 
         String dataSourceName = getAnnotationLiteralValue(annotationTypeReference, "name");
```
[the remark this answers is not in the document]
Comment (jonathanvila) [reply rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/datasource/DiscoverDataSourceAnnotationRuleProvider.java]: +1

File: rules-java-ee/tests/src/test/resources/discover-spring-remote-services/RestController.java (hunk 1 above)
Comment (jonathanvila) [reply rules-java-ee/tests/src/test/resources/discover-spring-remote-services/RestController.java]: +1

**Comment by jonathanvila** [comment]
Closed as the comments have been applied to the new [PR](https://github.com/windup/windup/pull/1340)

**Pull request closed without merging**